### PR TITLE
Refactor CLI docker save

### PR DESCRIPTION
**Pull request by HuKeping**
Signed-off-by: Hu Keping hukeping@huawei.com

### CODE REVIEW

File: api/client/save.go [hunk 1]
```diff
@@ -25,12 +25,13 @@ func (cli *DockerCli) CmdSave(args ...string) error {
...
+
 	if *outfile != "" {
-		output, err = os.Create(*outfile)
-		if err != nil {
+		if output, err = os.Create(*outfile); err != nil {
 			return err
 		}
-	} else if cli.isTerminalOut {
+	}
+	if *outfile == "" && cli.isTerminalOut {
```
Comment (cpuguy83) [inline api/client/save.go]: Can we bump this up before `if *outFile != ""`?

File: api/client/save.go [hunk 2]
```diff
@@ -44,14 +45,16 @@ func (cli *DockerCli) CmdSave(args ...string) error {
...
-			v.Add("names", arg)
-		}
-		if _, err := cli.stream("GET", "/images/get?"+v.Encode(), sopts); err != nil {
-			return err
-		}
+		return nil
 	}
+
+	v := url.Values{}
+	for _, arg := range cmd.Args() {
```
Comment (cpuguy83) [inline api/client/save.go]: It seems like we can get rid of the whole check before this block (for Args == 1) and just do this.
The `/images/get` and `/images/{name}/get` are the same handlers, I can't imagine them being any different at some point... honestly a very strange API.

**Comment by HuKeping** [comment]
updated @cpuguy83

**Comment by jfrazelle** [comment]
LGTM

**Comment by cpuguy83** [comment]
LGTM

**Pull request merged**